README.md

```diff
@@ -217,97 +211,99 @@ The `error` return is used to signal invalid Pattern
 structure, which could be bad UTF-8 or malformed JSON 
 or leaf values which are not provided as arrays.
 
-As many Patterns as desired can be added to a Matcher. 
-The `CoreMatcher` type does not support `DeletePattern()`
-but `pruner.Matcher` does.
+As many Patterns as desired can be added to a Quamina
+instance. More than one pattern can be added with the
+same `X` identifier.
 
 The `AddPattern` call is single-threaded; if multiple
 threads call it, they will block and execute sequentially.
-
 ```go
-func (m *Matcher) MatchesForJSONEvent(event []byte) ([]X, error)
+func (q *Quamina) DeletePatterns(x X) error 
 ```
-
-The `event` argument must be a JSON object encoded in
-correct UTF-8. 
+After calling this API, no list of matches from
+`AddPattern` will include the `X` value specified
+in the argument.
 
 The `error` return value is nil unless there was an
-error in the Event JSON.
+internal failure of Quamina’s storage system.
+```go
+func (q *Quamina) MatchesForEvent(event []byte) ([]X, error)
+```
+The `error` return value is nil unless there was an
+error in the encoding of the Event.
 
 The `[]X` return slice may be empty if none of the Patterns
-match the provided Event. 
+match the provided Event.
 
-```go
-func (m *Matcher) MatchesForFields([]Field) ([]X, error)
-```
-Performs the functions of `MatchesForJSON` on an 
-Event which has been flattened into a list of `Field`
-instances.  At the moment, `CoreMatcher` only returns
-an error if the `[]Field` argument is nil. `pruner.Matcher`
-can return an error if it suffers a failure in its 
-Pattern storage.
-
-These matching calls are thread-safe. Many threads may
-be executing it concurrently, even while `AddPattern` is
-also executing.  There is a significant performance 
-penalty if there is a high rate of `AddPattern` in
-combination with matching.
+### Concurrency
 
-```go
-func NewFJ(*Matcher) Flattener
-```
-Creates a new JSON-specific Flattener.
-```go
-func (fj *FJ) Flatten([]byte event) []Field
-```
-Transforms an event, which must be JSON object
-encoded in UTF-8, into a list of `Field` instances.
+A single Quamina instance can not safely be used by
+multiple goroutines at the same time.  However, the
+underlying data structure is designed for concurrent
+access and the `Copy` API is provided to support this.
 
 ```go
-func (fj *FJ) FlattenAndMatch([]byte event) ([]X, error)
+func (q *Quamina) Copy() *Quamina
```

Comment:
Is "Copy" the right name here? The code doesn't actually copy anything. (Same goes for `Flattener.Copy()`, I guess.) The word "copy" (along with not having read the actual implementation!) threw me off initially. Makes me think of a deep copy.

If I complain about "Copy", I should propose an alternative. Sadly I don't yet have a good one. In my mind it's something like `MatchingSession`, but that sounds terrible.